### DISCUSSION TITLE
Strange behaviour when trying to make an offline package from  Postgis layers in the cloud.

### DISCUSSION THREAD

**Discussion by ThomasDV1984** [opengisch/QField#3125] [Q&A]
I have a project stored on acugis consisting of a point layer "boom", a table without geometry "controle", that have a 1:n relation and a view "distinct" joining the 2 tables. I want the convert it to an offline project on qfield cloud, using the plugin. The upload seems succesfull.
![image](https://user-images.githubusercontent.com/77888654/181266126-086fa834-5c8d-4959-a1b7-efa11669d857.png)
I also added a secret 
![image](https://user-images.githubusercontent.com/77888654/181266855-a75bcf13-0ec8-4afa-862b-d31315bfc689.png)
But when I open the project in qfield I get the following error message and only the view is vissible in the project.
`[14:06, 28-07-2022] Thomas  Tablet: Project bb13f433-6b36-4aaa-803a-52c1f99a917a: Packaged layer `vta` is not valid. Error code invalid_dataprovider, error message: File "/Documents/vta.gpkg" missing.
[14:06, 28-07-2022] Thomas  Tablet: Project bb13f433-6b36-4aaa-803a-52c1f99a917a: Packaged layer `boomveiligheid` is not valid. Error code invalid_dataprovider, error message: File "/Documents/boomveiligheid.gpkg" missing.
[14:06, 28-07-2022] Thomas  Tablet: Failed to upload delta file, reason:
Error transferring https://app.qfield.cloud/api/v1/deltas/1d1c09be-fafb-456f-96a9-7d61d8afe703/ - server replied: Bad Request
[QF/object_not_found] Object not found
[14:06, 28-07-2022] Thomas  Tablet: Failed to upload delta file, reason:
Error transferring https://app.qfield.cloud/api/v1/deltas/1d1c09be-fafb-456f-96a9-7d61d8afe703/ - server replied: Bad Request
[QF/object_not_found] Object not found
[14:07, 28-07-2022] Thomas  Tablet: [QF/object_not_found] Object not found
[HTTP/400] https://app.qfield.cloud/api/v1/deltas/1d1c09be-fafb-456f-96a9-7d61d8afe703/ Server fout.{"code":"object_not_found","message":"Object not found"}
{"code":"object_not_found","message":"Object not found"}
Error transferring https://app.qfield.cloud/api/v1/deltas/1d1c09be-fafb-456f-96a9-7d61d8afe703/ - server replied: Bad Request
[14:07, 28-07-2022] Thomas  Tablet: Project bb13f433-6b36-4aaa-803a-52c1f99a917a: Packaged layer `boom` is not valid. Error code invalid_dataprovider, error message: File "/Documents/boom.gpkg" missing.
Project bb13f433-6b36-4aaa-803a-52c1f99a917a: Packaged layer `boom` is not valid. Error code invalid_dataprovider, error message: File "/Documents/boom.gpkg" missing.`

If I look at the offline project in qgis the path of both the layer "boom" and the layer "control" the one goes to documents and the other to downloads.
![image](https://user-images.githubusercontent.com/77888654/181267963-7850f7e8-1aca-4ab7-914e-aaa6b000b778.png)

A want to add that i have tested it with making the changes directly on the server and that worked fine.

**Comment by suricactus** [comment]
@ThomasDV1984 Your QGIS is most probably configured to store absolute paths. Please check if this is the case.

Also, in case you want QFieldCloud to connect to your pg database directly, you should chose the second radio option in QFieldSync when creating and uploading the project.

**Reply by ThomasDV1984** [reply]
The path is set to relative.
![image](https://user-images.githubusercontent.com/77888654/181773625-4c38ba39-1692-4b84-b015-996c813be0ed.png)

**Reply by ThomasDV1984** [reply]
And the layer "boom" I see in the qfieldcloud project created in qgis is not the same as the layer "boom" uploaded on the qfieldcloud server. The one on the server is the correct one but it cannot be opened on the app.

**Reply by suricactus** [reply]
Can you upload a test project for more detailed answer instead of shooting in the dark. Thanks!

**Reply by ThomasDV1984** [reply]
[boomcontrole.zip](https://github.com/opengisch/QField/files/9231426/boomcontrole.zip)
This is the project that is stored locally. If you need acces to my database I have to make a login.
In addition in this project is a geopackage called "VTA" it is not part of the project and comes from an other db but is syncs in anyway.

**Reply by ThomasDV1984** [reply]
Is this an issue or something for support?

**Reply by suricactus** [reply]
To be very honest I didn't manage to look at it yet, busy delivering the new QFieldCloud version. Sorry for the delay. I will try to have a look until the end of the month, in the mean time you can see if the new version magically solved your issue.

**Reply by suricactus** [reply]
> In addition in this project is a geopackage called "VTA" it is not part of the project and comes from an other db but is syncs in anyway.

Hm, it seems it's not only the `vta.gpkg` that is missing?

![image](https://user-images.githubusercontent.com/2820439/185215047-ac068ef2-f97b-426d-973e-4a7d182617d0.png)

**Comment by suricactus** [comment]
When you upload your project to the cloud, the error message you get is:

```
::<<<::cdc0f5cc-2192-45ba-b18d-2689d93b0a22 QGIS Layers Data
WARNING:QGIS_MSGLOG:Cannot open /Documents/boom.gpkg ().()
WARNING:QGIS_MSGLOG:Cannot open /Documents/boomveiligheid.gpkg ().()
WARNING:QGIS_MSGLOG:Cannot open /Documents/vta.gpkg ().()
WARNING:QGIS_MSGLOG:3 unavailable layer(s) found:
WARNING:QGIS_MSGLOG: * ../../../Documents/boom.gpkg
WARNING:QGIS_MSGLOG: * ../../../Documents/boomveiligheid.gpkg
WARNING:QGIS_MSGLOG: * ../../../Documents/vta.gpkg
INFO:root:QGIS project layer checks
Layer Name      Layer Id    Is Valid    Status                Error Summary                                   Provider Summary
--------------  ----------  ----------  --------------------  ----------------------------------------------  ------------------
OpenStreetMap   ...e20777   True        no_error
boom            ...17a990   False       invalid_dataprovider  File "/Documents/boom.gpkg" missing.
boomveiligheid  ...fedfb1   False       invalid_dataprovider  File "/Documents/boomveiligheid.gpkg" missing.
distinct        ...7f51cb   True        no_error
non_distinct    ...6132a9   True        no_error
vta             ...e25fcb   False       invalid_dataprovider  File "/Documents/vta.gpkg" missing.
```

Please make sure all your files are within the directory that holds the `.qgs` file. Also, please make sure all the layers are using these files, not those in `Downloads` or `Documents` directory.

In the meantime I will add this as a warning in QFieldSync.
